### ISSUE TITLE
Plaquette computation should compute spatial and temporal plaquettes separately

### ISSUE THREAD

**Issue by mikeaclark** [lattice/quda#305]
This should be straightforward to do.  :)

**Comment by nmrcardoso** [comment]
the file pgauge_plaquette.cu in quda feature gauge-fix branch
already does this.

**Comment by mikeaclark** [comment]
To reiterate our previous discussion, we need to get the two plaquette codes merged.  What are the outstanding issues preventing this?

**Comment by AlexVaq** [comment]
> To reiterate our previous discussion, we need to get the two plaquette codes merged. What are the outstanding issues preventing this?
> 
I agree, and really would like to do this, but I’m afraid I’m extremely busy at the moment: I’m going everyday to the hackathon in Lugano, and when this finishes I’ll have to prepare for the lattice conference, so I hope is not urgent.

I’ll put a note to work at the end of July on this, unless I can magically find time before.

**Comment by mikeaclark** [comment]
The need for this is quite urgent, as it's from someone's who just bought a cluster for doing QUDA and they want this asap  :)

The CPU code in this example computes the plaquettes whenever the gauge force is called and when this computation is off-loaded to QUDA, the plaquette values are not thus calculated.  I suggest therefore that for now I call Nuno's plaquette code in the gauge force and return these values.  This change should be minor (a dozen lines of code or so), so I'll just do it in the feature/gauge-fix branch.  When the plaqeutte codes are merged, this code will perhaps need a small update.

Sound reasonable?

**Comment by mathiaswagner** [comment]
You might also branch off from feature/gauge-fix and just do your changes there.
That gives us an easier path to either discard that temporary solution (in case we do have a better one in maybe one month) or just merge it in the gauge-fix branch if we want to keep it? 

May sound like overkill but on the other hand it is just a branch in git.

**Comment by AlexVaq** [comment]
Wait, I just had a look at the code and I think the modification is really simple. Maybe I can do it right now, but I won’t be able to test it thoroughly.

**Comment by mikeaclark** [comment]
That's ok.  If you make a branch with this in, I can test it when I plug it into the customer code.  Can you branch from release/0.7.x: since this is a minor change, we can include it in 0.7.2.

**Comment by AlexVaq** [comment]
Ok. I just realized that, since I calculate the plaquette in ascending order of coordinates, I don’t even have to introduce flow control in the kernels to separately evaluate the space-time plaquette.

I’ll adopt Nuno’s solution of storing the space-space placate in x and space-time in y of a double2.

**Comment by AlexVaq** [comment]
Well, I tested singleGPU and multiGPU with partition on one GPU only. I measured the plaquette in real configurations and compared with a precomputed result, and it agreed.

I should further test the spatial and temporal plaquettes. I mean, I didn't test those separately. A way to go is to compare with Nuno's code, if that has been tested. Another way to go is to write the CPU plaquette and solve these two issues in one stroke.
